File: medplum-java/medplum-model/src/main/java/com/fhir/Claim_SupportingInfo.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.Boolean;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableClaim_SupportingInfo.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface Claim_SupportingInfo {
  CodeableConcept category();

  Optional<String> valueString();

  Optional<CodeableConcept> reason();

  Optional<List<Extension>> extension();

  Optional<String> timingDate();

  Optional<Boolean> valueBoolean();

  Optional<Period> timingPeriod();

  Optional<List<Extension>> modifierExtension();

  Optional<Reference> valueReference();

  Optional<Quantity> valueQuantity();

  Optional<String> id();

  Optional<positiveInt> sequence();

  Optional<Attachment> valueAttachment();

  Optional<CodeableConcept> code();

  static ImmutableClaim_SupportingInfo.CategoryBuildStage builder() {
    return ImmutableClaim_SupportingInfo.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/Claim_SupportingInfo.ts
import { Attachment } from "./Attachment";
import { CodeableConcept } from "./CodeableConcept";
import { Extension } from "./Extension";
import { Period } from "./Period";
import { positiveInt } from "./positiveInt";
import { Quantity } from "./Quantity";
import { Reference } from "./Reference";

export interface Claim_SupportingInfo {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    sequence: positiveInt | null | undefined;
    category: CodeableConcept;
    code: CodeableConcept | null | undefined;
    timingDate: string | null | undefined;
    timingPeriod: Period | null | undefined;
    valueBoolean: boolean | null | undefined;
    valueString: string | null | undefined;
    valueQuantity: Quantity | null | undefined;
    valueAttachment: Attachment | null | undefined;
    valueReference: Reference | null | undefined;
    reason: CodeableConcept | null | undefined;
}
